tracer/src/Datadog.Trace/Propagators/SpanContextPropagator.cs

```diff
@@ -176,154 +228,63 @@ public void Inject<TCarrier>(SpanContext context, TCarrier carrier, Action<TCarr
             }
         }
 
-        /// <summary>
-        /// Extracts a <see cref="SpanContext"/> from its serialized dictionary.
-        /// </summary>
-        /// <param name="serializedSpanContext">The serialized dictionary.</param>
-        /// <returns>A new <see cref="SpanContext"/> that contains the values obtained from the serialized dictionary.</returns>
-        internal SpanContext? Extract(IReadOnlyDictionary<string, string?>? serializedSpanContext)
-        {
-            if (serializedSpanContext == null)
-            {
-                return null;
-            }
-
-            return Extract(serializedSpanContext, _readOnlyDictionaryValueGetterDelegate);
-        }
+        private record struct Key(string HeaderName, string TagPrefix);
```

Comment:
Should this be a `readonly record struct`? Also, not that this _isn't_ the same as 

```csharp
private readonly struct Key
{
    public string HeaderName;
    public string TagPrefix;
}
```

as it creates properties, in addition to the fields. If that's what you want, that's fine, just wanted to flag it

Reply:
Yes, I know is not the same, the code that it's replacing is this one: https://github.com/DataDog/dd-trace-dotnet/blob/master/tracer/src/Datadog.Trace/SpanContextPropagator.cs#L277-L320

I have problems using `readonly` :
![image](https://user-images.githubusercontent.com/69803/157446413-fa7c73aa-ce65-49d4-aaaf-f2b3c67f9af4.png)

Reply:
Hmmm, won't we get extra copies by it not being readonly? 🤔

Reply:
Yes, I'm thinking if a defensive copy here can be an issue. So I'm thinking on play safe here and restore the old readonly struct.

Reply:
done in [a403f89](https://github.com/DataDog/dd-trace-dotnet/pull/2536/commits/a403f89e0997f8ac2a866d550e15fc07a65cd672)